Tests/MapboxNavigationTests/VanishingRouteLineTests.swift

```diff
@@ -0,0 +1,262 @@
+        XCTAssertEqual(routePoints.flatList[128].longitude, routePoints.flatList[129].longitude, accuracy: 0.000001)
+    }
+    
+    func testUpdateUpcomingRoutePointIndex() {
+        // https://github.com/mapbox/mapbox-navigation-android/blob/0ca183f7cb7bec930521ea9bcd59d0e8e2bef165/libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApiTest.kt#L802
+        let route = getRoute()
+        
+        navigationMapView.initPrimaryRoutePoints(route: route)
+        navigationMapView.routeLineGranularDistances = nil
+        XCTAssertEqual(navigationMapView.fractionTraveled, 0.0)
+        
+        let routeProgress = getRouteProgress()
+        
+        navigationMapView.updateUpcomingRoutePointIndex(routeProgress: routeProgress)
+        
+        // Because mapbox-directions-swift parses the route with one more duplicate coordinate in the last step of each route leg. The one leg route has one more coordinate compared with Android.
+        XCTAssertEqual(navigationMapView.routeRemainingDistancesIndex, 7)
+    }
+    
+    func testUpdateUpcomingRoutePointIndexWhenPrimaryRoutePointsIsNill() {
+        // https://github.com/mapbox/mapbox-navigation-android/blob/0ca183f7cb7bec930521ea9bcd59d0e8e2bef165/libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApiTest.kt#L846
+        let routeProgress = getRouteProgress()
+        
+        navigationMapView.updateUpcomingRoutePointIndex(routeProgress: routeProgress)
+        XCTAssertNil(navigationMapView.routeRemainingDistancesIndex)
+    }
+    
+    func testUpdateFractionTraveled() {
+        // https://github.com/mapbox/mapbox-navigation-android/blob/0ca183f7cb7bec930521ea9bcd59d0e8e2bef165/libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApiTest.kt#L636
+        let route = getRoute()
+        let routeProgress = getRouteProgress()
+        
+        let coordinate = route.shape!.coordinates[1]
+        navigationMapView.routeLineTracksTraversal = true
+        navigationMapView.show([route])
+        navigationMapView.updateUpcomingRoutePointIndex(routeProgress: routeProgress)
+        navigationMapView.updateFractionTraveled(coordinate: coordinate)
+        
+        let expectedFractionTraveled = 0.3240769449298392
+        XCTAssertEqual(navigationMapView.fractionTraveled, expectedFractionTraveled, accuracy: 0.0000000001)
+    }
+    
+    func testUpdateRouteLineWithDifferentDistance() {
+        let route = getRoute()
+        let routeProgress = getRouteProgress()
+        let coordinate = route.shape!.coordinates[1]
+        
+        navigationMapView.routes = [route]
+        navigationMapView.routeLineTracksTraversal = true
+        navigationMapView.show([route], legIndex: 0)
+        navigationMapView.updateUpcomingRoutePointIndex(routeProgress: routeProgress)
+        setUpCameraZoom(at: 5.0)
```

Comment:
`testUpdateRouteLineWithDifferentDistance` passes even without setting zoom level, do we need it here? I'd also add brief comment on what exactly you do in each test. At this point it's hard to follow some of them.

Reply:
Thanks for the suggestion, I'll add the comment and documentation for each test. The default zoom level is smaller than `1` which could pass the test.